### PR TITLE
Poker: Convert to structured answer

### PR DESCRIPTION
**Pull request by kirkins**
## Description of new Instant Answer, or changes

- Converts poker goodie to structured answer
- Replace tabs with spaces
- Add negative tests

## Related Issues and Discussions

#1163

## People to notify

@moollaza @mintsoft 

## Testing & Review
To be completed by Community Leader (or DDG Staff) when reviewing Pull Request

**Pull Request**
- [ ] Title follows correct format (Specifies Instant Answer + Purpose)
- [ ] Description contains a valid Instant Answer Page Link (e.g. https://duck.co/ia/view/my_ia)

**Instant Answer Page** (for new Instant Answers)
- [ ] Instant Answer page is correctly filled out and contains:
    - The **Title** is appropriately named and formatted
    - The **IA topics** are present and appropriate
    - The **Description** is clear and coherent
    - **Source Name** exists if applicable
    - All **Example Queries** trigger on [Beta](https://beta.duckduckgo.com/)

**Code**
- [ ] Adheres to the [DuckDuckGo Style Guide](https://docs.duckduckhack.com/resources/code-style-guide.html)
- [ ] Behaviour is appropriately tested. If improvement, tests are adequately extended.
- [ ] There is no unnecessary files in place (such as editor config files)
- [ ] There is no API keys / secrets present
- [ ] Tests are passing (run `$ duckpan test <goodie_id>`)
    - Tester should report any failures

**Ready to merge?**

- [ ] Has this IA been deployed to and tested on [beta.duckduckgo.com](https://beta.duckduckgo.com/)?
- [ ] For larger commits, has this been approved be more than one community member?
- [ ] The number of reviews is appropriate for this type of PR
- [ ] The commit message is clear, coherent and fitting

**Pull Request Review Guidelines**: https://docs.duckduckhack.com/programming-mission/pr-review.html

---

Instant Answer Page: https://duck.co/ia/view/poker

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: lib/DDG/Goodie/Poker.pm [hunk 1]
```diff
@@ -79,14 +79,25 @@ handle remainder => sub {
...
-        'frequency'   => [$freq, 'html' => "$freq $link"],
-        'probability' => [$prob, 'html' => "$prob $link"],
+        'odds'        => $odds,
+        'frequency'   => $freq,
+        'probability' => $prob,
     );
 
-    return @{$answer{$query}} if exists $answer{$query};
+    my %statistic = (
+        'odds'        => $odds{$hand}." : 1",
```
Comment (moollaza) [inline lib/DDG/Goodie/Poker.pm]: FYI: no need for the single quotes here 👍

File: lib/DDG/Goodie/Poker.pm [hunk 2]
```diff
@@ -79,14 +79,25 @@ handle remainder => sub {
...
+        'probability' => $probability{$hand}."%",
+    );
+
+    my $string_answer = $answer{$query};
+    my $structured_answer = {};
+    $structured_answer->{templates}->{group} = 'text';
+    $structured_answer->{data}->{title} = $statistic{$query};
+    $structured_answer->{data}->{subtitle} = $answer{$query};
+
+    return $string_answer, structured_answer => $structured_answer;
```
Comment (moollaza) [inline lib/DDG/Goodie/Poker.pm]: nit: I would just define the hash, and string result here, like we typically do. Then we have two less variables to worry about:

```perl
return $answer{$query}, structured_answer => {
    templates => {
        group => 'text'
    },
    data => {
        title => $statistic{$query},
        subtitle => $answer{$query}
    }
}
```

File: lib/DDG/Goodie/Poker.pm [hunk 3]
```diff
@@ -79,14 +79,25 @@ handle remainder => sub {
...
+        'odds'        => $odds{$hand}." : 1",
+        'frequency'   => $frequency{$hand}." out of 2,598,960",
+        'probability' => $probability{$hand}."%",
+    );
+
+    my $string_answer = $answer{$query};
+    my $structured_answer = {};
+    $structured_answer->{templates}->{group} = 'text';
+    $structured_answer->{data}->{title} = $statistic{$query};
+    $structured_answer->{data}->{subtitle} = $answer{$query};
```
Comment (moollaza) [inline lib/DDG/Goodie/Poker.pm]: Repeating the answer in the subtitle is a little redundant.

The subtitle usually just rephrases the query to provide context for the answer in the title

**Comment by bsstoner** [comment]
This was released to production 🚀

**Pull request merged**